contracts/cheddar_5x5_tic_tac_toe/src/callbacks.rs

```diff
@@ -33,4 +33,15 @@ impl Contract {
             self.stats.insert(&user.clone(), &stats);
         }
     }
+    #[private]
+    pub fn cheddar_withdraw_callback(&mut self, user: &AccountId, vault: Vault) {
+        if promise_result_as_failed() {
+            log!(
+                "FT withdrawal failed. Recovering state for {} account",
+                user.clone(),
+            );
+            self.registered_players.insert(user, &vault);
```

Comment:
why do we need to re-insert? What if there was a change in the meantime. I think we should update the vault (so, in the argument set the amount we need to re-add) rather than potentially copy of a vault.

Also, cheddar withdraw != user unsubscription (unregistration)  from the game. As a user a may still keep playing and decide to withdraw some cheddar.

Reply:
As far as I am concerned we cannot modify values inside of the `UnorderedMap` we can only read it inside of the callback and re-insert it. I updated it. Let me know if its ok now

Yes the unregister is different from withdraw_cheddar in the withdraw we only transfer back the requested amount of cheddar. The user stays registered for the game.